src/tests-rosetta/construction.ts

```diff
@@ -0,0 +1,2492 @@
+import * as assert from 'assert';
+import {
+  AnchorMode,
+  AuthType,
+  bufferCV,
+  ChainID,
+  createStacksPrivateKey,
+  getPublicKey,
+  makeSTXTokenTransfer,
+  makeUnsignedContractCall,
+  makeUnsignedSTXTokenTransfer,
+  noneCV,
+  pubKeyfromPrivKey,
+  publicKeyToString,
+  SignedTokenTransferOptions,
+  someCV,
+  standardPrincipalCV,
+  TransactionSigner,
+  tupleCV,
+  uintCV,
+  UnsignedContractCallOptions,
+  UnsignedTokenTransferOptions,
+} from '@stacks/transactions';
+import * as BN from 'bn.js';
+import { StacksCoreRpcClient } from '../core-rpc/client';
+import { bufferToHexPrefixString, timeout } from '../helpers';
+import {
+  RosettaConstructionCombineRequest,
+  RosettaConstructionCombineResponse,
+  RosettaAccountIdentifier,
+  RosettaConstructionDeriveRequest,
+  RosettaConstructionDeriveResponse,
+  RosettaConstructionHashRequest,
+  RosettaConstructionHashResponse,
+  RosettaConstructionMetadataRequest,
+  RosettaConstructionParseRequest,
+  RosettaConstructionParseResponse,
+  RosettaConstructionPayloadsRequest,
+  RosettaConstructionPreprocessRequest,
+  RosettaConstructionPreprocessResponse,
+  RosettaConstructionMetadataResponse,
+} from '@stacks/stacks-blockchain-api-types';
+import {
+  getRosettaNetworkName,
+  RosettaConstants,
+  RosettaErrors,
+  RosettaErrorsTypes,
+} from '../api/rosetta-constants';
+import { getStacksTestnetNetwork, testnetKeys } from '../api/routes/debug';
+import {
+  getSignature,
+  getStacksNetwork,
+} from '../rosetta-helpers';
+import { makeSigHashPreSign, MessageSignature } from '@stacks/transactions';
+import { decodeBtcAddress } from '@stacks/stacking';
+
+
+describe('Rosetta API', () => {
+  let db: PgDataStore;
+  let client: PoolClient;
+  let eventServer: Server;
+  let api: ApiServer;
+
+  function standByForTx(expectedTxId: string): Promise<DbTx> {
+    const broadcastTx = new Promise<DbTx>(resolve => {
+      const listener: (txId: string) => void = async txId => {
+        const dbTxQuery = await api.datastore.getTx({ txId: txId, includeUnanchored: true });
+        if (!dbTxQuery.found) {
+          return;
+        }
+        const dbTx = dbTxQuery.result as DbTx;
+        if (
+          dbTx.tx_id === expectedTxId &&
+          (dbTx.status === DbTxStatus.Success ||
+            dbTx.status === DbTxStatus.AbortByResponse ||
+            dbTx.status === DbTxStatus.AbortByPostCondition)
+        ) {
+          api.datastore.removeListener('txUpdate', listener);
+          resolve(dbTx);
+        }
+      };
+      api.datastore.addListener('txUpdate', listener);
+    });
+
+    return broadcastTx;
+  }
+
+  async function standByForPoxToBeReady(): Promise<void>{
+    let tries = 0;
+    while(true){
+      try{
+        tries++;
+        await  new StacksCoreRpcClient().getPox();
+        return Promise.resolve();
+      }
+      catch(error){
+        console.log('Error getting pox info on try ' + tries, error);
+        timeout(100);
+      }
+    }
+  }
```

Comment:
Echoing what we discussed on the call, this function could be improved to wait for PoX in a robust way